py-polars/polars/io/_utils.py

```diff
@@ -16,7 +17,7 @@ def _is_glob_pattern(file: str) -> bool:
 
 
 def _is_supported_cloud(file: str) -> bool:
-    return file.startswith(("s3", "gs", "az", "adl", "abfs"))
+    return bool(re.match("^(s3a?|gs|gcs|file|abfs|abfss|azure|az|adl)://", str(file)))
```

Comment:
I added this explicit cast to str here because you can also pass in a Path. I'll have a look if it would still work (or not)

Reply:
```
>>> from pathlib import Path
>>> p = Path('/tmp/bumba.jpg')
>>> import re
>>> re.match("^(s3a?|gs|gcs|file|abfs|abfss|azure|az|adl)://", p)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\APPS\Anaconda3\envs\consumer_feed\Lib\re\__init__.py", line 166, in match
    return _compile(pattern, flags).match(string)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: expected string or bytes-like object, got 'WindowsPath'
```

So the explicit cast is really necessary because we could get a `Path` in here.

Reply:
I know the type hint says str only, but:

pl.scan_parquet is located in: polars/io/parquet/functions/scan_parquet -- `source: str | Path,`

which calls: polars/lazyframe/frame/_scan_parquet -- `source: str,`

which calls: _is_supported_cloud

So we could be getting a Path inside. It's just not correctly type hinted.

Reply:
In that case, the type hint must be updated indeed!

Reply:
Sorry, when digging into it I saw this:
```python
    if isinstance(source, (str, Path)):
        source = normalize_filepath(source)
```
Which always returns `str`. So I spoke too soon. It'll always be a `str` and the type hinting is accurate.
Removing the str-cast instead.